numba/core/typing/templates.py

```diff
@@ -402,6 +402,15 @@ class CallableTemplate(FunctionTemplate):
     def apply(self, args, kws):
         generic = getattr(self, "generic")
         typer = generic()
+        try:
+            match_sig = inspect.signature(typer)
```

Comment:
i don't think it should be catching `TypeError` from `.signature`

```suggestion
        match_sig = inspect.signature(typer)
        try:       
```

Reply:
I guess this can only come from the arg to `signature` being non-callable, which would indicate a more fundamental problem, will make the change, thanks!

Reply:
d63dab5